e2e/tests/keycloak-openmrs-flows.spec.ts

```diff
@@ -1,191 +1,183 @@
 import { test, expect } from '@playwright/test';
-import { HomePage, delay, randomKeycloakRoleName } from '../utils/functions/testBase';
-import { randomOpenMRSRoleName, randomSupersetRoleName } from '../utils/functions/testBase';
+import { Keycloack, randomKeycloakRoleName } from '../utils/functions/keycloak';
+import { OpenMRS, delay, randomOpenMRSRoleName } from '../utils/functions/openmrs';
+import { Superset, randomSupersetRoleName} from '../utils/functions/superset';
 import { O3_URL, KEYCLOAK_URL } from '../utils/configs/globalSetup';
 
-let homePage: HomePage;
+let openmrs: OpenMRS;
+let keycloack: Keycloack;
+let superset: Superset;
 
 test.beforeEach(async ({ page }) => {
-  homePage = new HomePage(page);
+  openmrs = new OpenMRS(page);
+  keycloack = new Keycloack(page);
+  superset = new Superset(page);
+
+  await openmrs.initiateLogin();
+  await expect(page).toHaveURL(/.*home/);
 });
 
 test('Creating an OpenMRS role creates the corresponding Keycloak role.', async ({ page }) => {
   // replay
-  await homePage.initiateLogin();
-  await expect(page).toHaveURL(/.*home/);
   await page.goto(`${O3_URL}/openmrs/admin/users/role.list`);
-  await homePage.addOpenMRSRole();
+  await openmrs.addRole();
 
   // verify
-  await homePage.goToKeycloak();
+  await keycloack.open();
   await expect(page).toHaveURL(/.*console/);
-  await homePage.goToClients();
-  await homePage.selectOpenMRSId()
+  await keycloack.goToClients();
+  await keycloack.selectOpenMRSId()
```

Comment:
Shouldn't it be more something like `selectOpenMRSClient()` to remain in Keycloak's lingo? Or this is UI-driven and there is literally something called ID on the screen?

Reply:
Actually` selectOpenMRSClient() ` would imply something different however, this is UI-driven with a label called client Id on the screen.